go/cmd/betabot/main.go

```diff
@@ -61,22 +63,28 @@ func betabot() error {
 	{
 		s, err := messengerClient.EventStream(ctx, &bertymessenger.EventStream_Request{})
 		if err != nil {
+			log.Printf("Failed to listen EventStream %v", err)
 			return err
 		}
 
-		handlerMutex := sync.Mutex{}
+		var wg sync.WaitGroup
+
 		go func() {
 			for {
-				gme, err := s.Recv()
-				if err != nil {
-					log.Printf("stream error: %v", err)
-				}
-				handlerMutex.Lock()
-				err = handleEvent(ctx, messengerClient, gme)
-				handlerMutex.Unlock()
-				if err != nil {
-					log.Printf("event handler error: %v", err)
+				select {
```

Comment:
The goal of the select is to wait multiples channel (you should maybe `<-s.Recv()` (doesn't work right now)) :
```go
defer wg.Done()
// Loops and ...
select {
case gme, err := <-s.Recv():
  // handle event
case <-ctx.Done():
  return
}
```